samples/solutions/provision_symmetric_key_failover_device_delete.py

```diff
@@ -0,0 +1,374 @@
+                await asyncio.sleep(MONITOR_TIME_BETWEEN_SUCCESS_ASSIGNMENTS)
+            elif self.iothub_assignment_fail_event.is_set():
+                try:
+                    symmetric_key = os.getenv("PROVISIONING_SYMMETRIC_KEY")
+                    provisioning_client = await self.create_dps_client(symmetric_key)
+                    self.log_info_and_print("Registering the device...")
+                    registration_result = await provisioning_client.register()
+                    print("The complete registration result is")
+                    print(registration_result.registration_state)
+                    if registration_result.status == "assigned":
+                        self.log_info_and_print(
+                            "Will create hub client to send telemetry from the provisioned device"
+                        )
+                        await self.create_hub_client(registration_result, symmetric_key)
+                        self.iothub_assignment_sucess_event.set()
+                        self.iothub_assignment_fail_event.clear()
+                        self.log_error_and_print(
+                            "Registration was done and device was assigned correctly to an "
+                            "IoTHub via the registration process.Will check if all is right "
+                            "after some time..."
+                        )
+                        await asyncio.sleep(MONITOR_TIME_BETWEEN_SUCCESS_ASSIGNMENTS)
+                    else:
+                        self.log_error_and_print(
+                            "Registration was done but device was not assigned correctly to an "
+                            "IoTHub via the registration process.Will try registration "
+                            "again after some time..."
+                        )
+                        await asyncio.sleep(SLEEP_TIME_BETWEEN_CHECKING_REGISTRATION)
+                        self.iothub_assignment_fail_event.set()
+                        self.iothub_assignment_sucess_event.clear()
+                except Exception as e:
+                    self.log_error_and_print(
+                        "Registration process failed because of error {}".format(get_type_name(e))
+                    )
+                    raise Exception(
+                        "Caught an unrecoverable error that needs to be "
+                        "fixed from user end while registering. Will exit application..."
+                    )
+                finally:
+                    await provisioning_client.shutdown()
+
+            if self.exit_app_event.is_set():
+                return
+
+    async def wait_for_connect_and_send_telemetry(self):
+        id = 1
+        while True:
+            if not self.iothub_client:
+                # Time to check if device has been provisioned
+                self.log_info_and_print(
+                    "IoTHub client is still nonexistent for telemetry. "
```

Comment:
maybe better to say:
IoTHub client is not assigned. Telemetry operation failed.